ads/inabox/inabox-messaging-host.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PositionObserver} from './position-observer';
+import {dev} from '../../src/log';
+
+/** @const */
+const TAG = 'inabox-messaging-host';
+/** @const */
+const AMP_MESSGE_PREFIX = 'amp-';
+/** @const */
+const REQUEST_TYPE = {
+  SEND_POSITIONS: 'send-positions',
+};
+/** @const */
+const RESPONSE_TYPE = {
+  POSITION: 'position',
+};
+
+export class InaboxMessagingHost {
+
+  constructor(win, iframes) {
+    this.win_ = win;
+    this.iframes_ = iframes;
+    this.iframeMap_ = {};
```

Comment:
As per Justin's comments on my PR, there is a map() helper function that apparently you are supposed to use for this